File: solutions/r/darts/darts.R
score <- function(x, y) {
  dist <- sqrt(x^2 + y^2)

  if (dist <= 1) {
    10
  } else if (dist <= 5) {
    5
  } else if (dist <= 10) {
    1
  } else {
    0
  }
}

File: solutions/fsharp/darts/Darts.fs
module Darts

let (|RadiusGreaterThan|) r (x, y) : bool = sqrt (x ** 2.0 + y ** 2.0) > r

let score (x: double) (y: double) : int =
    match x, y with
    | RadiusGreaterThan 10.0 true -> 0
    | RadiusGreaterThan 5.0 true -> 1
    | RadiusGreaterThan 1.0 true -> 5
    | _ -> 10
